opencart1.5.x/upload/catalog/controller/payment/xendit.php

```diff
@@ -79,10 +79,10 @@ public function process_notification() {
             $this->load->model('payment/xendit');
             $this->load->model('checkout/order');
 
-            $response = json_decode(file_get_contents('php://input'), true);
-            $invoice_id = $response['id'];
-            $external_id = $response['external_id'];
-            $order_id = str_replace(self::EXT_ID_PREFIX, "", $external_id);
+            $original_response = json_decode(file_get_contents('php://input'), true);
+            $invoice_id = $original_response['id'];
+            $external_id = $original_response['external_id'];
+            $order_id = preg_replace('/[^0-9]/', '', $external_id);
```

Comment:
@candrasaputra for your previous concern, I have change it to extract numbers from string, so it doesn't depend on the prefix format anymore.

Reply:
whoa this is nice, we should use this in the future